utils/compare/graph.py

```diff
@@ -0,0 +1,306 @@
+    ids = "\" OR uuid: \"".join(uuids)
+    query = {
+        "query": {
+            "query_string": {
+                "query": (
+                    f'( uuid: \"{ids}\" )'
+                    f' AND metricName: "jobSummary"'
+                    )
+            }
+        }
+    }
+    #print(query)
+    es = ElasticService(index=index)
+    response = es.post(query)
+    es.close()
+    runs = [item['_source'] for item in response["hits"]["hits"]]
+    return runs
+
+def processData(data: dict, columns: list[str], func="mean") :
+    #pprint.pprint(data)
+    df = pd.json_normalize(data)
+    filterDF = burnerFilter(df, columns)
+    if func == "avg" or func == "mean": 
+        ptile = filterDF.groupby(columns[:-1])[columns[-1]].mean()
+    elif func == "max": 
+        ptile = filterDF.groupby(columns[:-1])[columns[-1]].max()
+    return ptile
+
+def processBurner(data: dict) :
+    #pprint.pprint(data)
+    df = pd.a(data)
+    filterDF = burnerFilter(df)
+    #print('filter df ' + str(filterDF))
+    ptile = filterDF.groupby(['quantileName'])['P99'].mean()
+    #print("p tile" + str(ptile))
+    return ptile
+
+def jobFilter(pdata: dict, data: dict):
+    columns = ['uuid','jobConfig.jobIterations']
+    pdf = pd.json_normalize(pdata)
+    pick_df = pd.DataFrame(pdf, columns=columns)
+    iterations = pick_df.iloc[0]['jobConfig.jobIterations']
+    df = pd.json_normalize(data)
+    ndf = pd.DataFrame(df, columns=columns)
+    ids_df = ndf.loc[df['jobConfig.jobIterations'] == iterations ]
+    return ids_df['uuid'].to_list()
+
+def burnerFilter(data: dict, columns: list[str]) :
+    #
+    # Filter out aspects of the test to norm results
+    #
+    #pprint.pprint(data)
+    ndf = pd.DataFrame(data, columns=columns)
+    print("ndf " + str(ndf))
+    return ndf
+
+def getResults(uuid: str, uuids: list, index_str: str, metrics: dict ):
+
+    if len(uuids) > 1 and uuid in uuids:
+        uuids.remove(uuid)
+    ids = "\" OR uuid: \"".join(uuids)
+    #print(ids)
+    metric_string = ""
+    for k, v in metrics.items(): 
+        metric_string += f' AND {k}: {v}'
+
+    
+    #print('get results ' + str(ids))
+    query = {
+        "query": {
+            "query_string": {
+                "query": (
+                    f'( uuid: \"{ids}\" )' + metric_string
+                    )
+            }
+        }
+    }
+    print(query)
+    print('index ' + str(index_str))
+    es = ElasticService(index=index_str)
+    response = es.post(query)
+    #print('respnse ' + str(response))
+    es.close()
+    runs = [item['_source'] for item in response["hits"]["hits"]]
+    return runs
+
+def getMatchRuns(meta: dict, workerCount: False, previous_version=False):
+    index = "perf_scale_ci"
+    version = meta["ocpVersion"][:4]
+    if previous_version: 
+        version = float(version) - .01
+    
+    # https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-range-query.html
+    if not workerCount: 
```

Comment:
```suggestion
        query_params = {
        "benchmark": meta["benchmark"],
        "workerNodesType": meta["workerNodesType"],
        "masterNodesType": meta["masterNodesType"],
        "networkType": meta["networkType"],
        "platform": meta["platform"],
        "ocpVersion": f"{version}*",
        "jobStatus": "success",
    }

    if workerCount:
        query_params.update({
            "masterNodesCount": meta["masterNodesCount"],
            "workerNodesCount": meta["workerNodesCount"],
        })

    query = {
        "_source": ["uuid"],
        "query": {
            "query_string": {
                "query": " AND ".join([f'{key}: "{value}"' for key, value in query_params.items()]),
            }
        }
    }

    es = ElasticService(index=index)
    response = es.post(query)
    es.close()
    return [item['_source']['uuid'] for item in response["hits"]["hits"]]
```
This will only fetch the `uuid` portion of the docs instead of the entire docs in-memory.